google-summer-of-code-2024.md

```diff
@@ -0,0 +1,323 @@
+|   |
+│   ├── Cypress               // contains cypress configuration for frontend testing
+│   ├── nginx                // contains nginx server configurations
+│   ├── node_modules        // contains all dependencies needed to run and build the UI.
+│   ├── public             // Contains public pages and assets for the project
+|   ├── src               // Contains all UI source folders holding pags and react scripts of the project  
+|   ├── Dockerfile          // Contains the commands and procedures to build docker images of the project.
+|   ├── Dockerfile.prod     // contains neccessary commands and procedures to build base prod docker images for the project.
+|   ├── package.json        // provide dependencies management , build scripts , project meta data dependency versions
+    ├── package-lock.json
+|    
+|    
+├──src     // this folder holds the actual java backend logic including services, controllers, DAOs , entities , tests,  etc.
+|    |   
+|    └── main 
+|    |     └── Filtered resources   // This contains build.property file which specifies the project version
+|    │     └── Java    // Contains packages which defines backend logic
+|    |     └── resources     // Contains resources such as persistence , reports and static configurations.
+|    |     └── webapp     // Contains formal Java server pages based UI which is being rewritten to React,js
+|    │ 
+|    ├── test    // This folder holds unit and  integration  tests of the logic.
+
+|    |
+|    |
+|    |
+|    |
+├── Volumes       // This folder holds volumes with database, nginx and tomcat configurations.
+
+└── Dockerfile    // contains neccessary commands and procedures to build base prod docker images for the project.
+└── Docker-compose.yml  // contains neccessary commands and  steps to build multiple services.
+└── .gitmodules  // contains different git modules such as data export
+└── pom.xml    // contains project information < artifacts and group id of the project >, dependencies , plugins and build configurations
+```
+
+## Technologies and development tools.
+
+### Technologies
+
+OpenELIS Global 2 is built upon a robust technology stack:
+
+* Backend: [Java 11](https://www.oracle.com/java/technologies/downloads/)
+* Database: [PostgreSQL](https://www.postgresql.org/)
+* Frontend: [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript) and [React](https://legacy.reactjs.org/docs/getting-started.html)
+* Framework: [Spring](https://spring.io/)
+
+
+### Ensuring Code Quality
+
+
+We prioritize maintaining high code quality through unit, integration, and end-to-end tests. OpenELIS utilizes various testing tools, including:
+
+* JUnit 5 (https://junit.org/junit5/docs/current/user-guide/)
+* Mockito (https://site.mockito.org/)
+* TestNG (https://testng.org/)
+* Cypress (https://www.cypress.io/)
+
+
+It is high recommended to rebuild the application and make sure your test cases are passing before opening any pull request.
```

Comment:
"It is **high** recommended....." should be "It is **highly** recommended........."